pkg/gofr/metrics/register.go

```diff
@@ -114,13 +114,22 @@ func (m *metricsManager) NewHistogram(name, desc string, buckets ...float64) {
 	}
 }
 
+// Developer Note: float64Gauge is used instead of metric.Float64ObservableGauge because we need a synchronous gauge metric
+// and otel/metric supports only asynchronous (Float64ObservableGauge) metric.
+// And if we use the otel/metric, we would not be able to have support for labels, Hence created a custom type to implement it.
+type float64Gauge struct {
```

Comment:
we can place all the custom types declared at the start of the file where other structs and interfaces are there.